### PR TITLE
ci/add e2e tests report visualisation [QA-518]

### PR DESCRIPTION
**Pull request by Karamann**
# Orfium GitHub Actions

## Description

- Add test report viewer in e2e tests reusable action (example https://github.com/Orfium/universal-catalog-mapper/actions/runs/4894297717/jobs/8738554725)

## Checklist

- [x] I have reviewed my code and checked that there are no unrelated changes in this pull request
- [x] I have created a JIRA ticket describing the purpose of this pull request
- [x] I have checked if the changes to the templates are breaking or have contacted the DevOps team for feedback
- [x] I have updated the template with the Actionlint Format guidelines: [Actionlint](https://github.com/rhysd/actionlint#readme). Information on how to do that are provided on the CONTRIBUTING.md file under docs/ folder.

### CODE REVIEW

File: .github/workflows/test-e2e.yml [hunk 1]
```diff
@@ -79,6 +89,15 @@ jobs:
           TESTRAIL_PASSWORD: ${{ secrets.TESTRAIL_PASSWORD }}
         run: make testrail-results
 
+      - name: visualise-e2e-test-reports
+        if: always()
```
Comment (Aloutsis) [inline .github/workflows/test-e2e.yml:93]: Why do we need just an "if: always()" here?
We may want this step to run if some of the previous steps fail, but not if none of them do ( e.g. checkout ), there shouldn't be a reason to. I would recommend to add ids to the specific steps that we want to run or not and then check run status on those steps. The way to do is described here: https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context

File: .github/workflows/test-e2e.yml [hunk 2]
```diff
@@ -29,6 +35,10 @@ env:
 jobs:
   end-to-end-api-tests:
     runs-on: ubuntu-latest
+    permissions:
```
Comment (Aloutsis) [inline .github/workflows/test-e2e.yml:38]: Why are those permissions needed? I didn't see anything related to the dorny/test-reporter GitHub page.

Comment (Karamann) [reply .github/workflows/test-e2e.yml:38]: Yes it is essential but the dudes have not updated the README! 🤦🏼 : https://github.com/dorny/test-reporter/issues/229#fromHistory

**Review by geokats7 — APPROVED** [review]

File: .github/workflows/test-e2e.yml [hunk 3]
```diff
@@ -79,6 +91,15 @@ jobs:
           TESTRAIL_PASSWORD: ${{ secrets.TESTRAIL_PASSWORD }}
         run: make testrail-results
 
+      - name: visualise-e2e-test-reports
+        if: steps.e2e-api-tests.outcome != 'skipped' || steps.e2e-ui-tests.outcome != 'skipped'
+        uses: dorny/test-reporter@v1
+        with:
+          name: E2E Test Reports
+          path: ${{ github.workspace }}/${{ inputs.API_TEST_RESULTS_PATH }},${{ github.workspace }}/${{ inputs.UI_TEST_RESULTS_PATH }} # Coma separated list of paths to test results, all matched result files must be of the same format
```
Comment (constantakopoulos) [inline .github/workflows/test-e2e.yml:99]: comma*

**Review by constantakopoulos — APPROVED** [review]

**Review by Aloutsis — APPROVED** [review]
Looks good.

**Pull request merged**